programs/stake/src/stake_state.rs

```diff
@@ -1023,6 +1057,34 @@ impl MergeKind {
     }
 }
 
+/// Calculate the effective credits observed for two stakes whenn merging
+///
+/// When merging two activating or active stakes, the credits observed of the
+/// merged stake is the weighted average of the two stakes' credits observed.
+fn stake_weighted_credits_observed(
+    stake: &Stake,
+    merge_lamports: u64,
+    merge_credits_observed: u64,
```

Comment:
As you noticed, I went with these terms because I get tripped up by `source_*` every time I read through the code. I like `lamports_to_merge` though! Going off of https://www.dictionary.com/browse/merge, how about `absorbed_*`?  so `absorbed_stake`, `absorbed_lamports` and `absorbed_credits_observed`?

Reply:
I like absorbed! Seems a lot more clear which stake acct is disappearing into the other.
Even though it adds churn, can we update on the calling side as well? Maybe a separate renaming PR so that the functionality changes stay distinct?

Reply:
Sounds good, I can update the variable names in the new functions, and rename everywhere in a separate PR